### PR TITLE
feat: add animated banner guild feature

### PR DESCRIPTION
**Pull request by iShibi**
**Please describe the changes this PR makes and why it should be merged:**

- discord/discord-api-docs#3948

**Status and versioning classification:**

### CODE REVIEW

File: packages/discord.js/src/structures/Guild.js [hunk 1]
```diff
@@ -512,6 +513,15 @@ class Guild extends AnonymousGuild {
...
 
+  /**
+   * The URL to this guild's banner.
+   * @param {ImageURLOptions} [options={}] Options for the Image URL
+   * @returns {?string}
+   */
+  bannerURL({ format, size, dynamic } = {}) {
+    return this.banner && this.client.rest.cdn.Banner(this.id, this.banner, format, size, dynamic);
+  }
+
```
Comment (Jiralite) [inline packages/discord.js/src/structures/Guild.js:524]: This should instead be modified in the `AnonymousGuild` class instead:

https://github.com/discordjs/discord.js/blob/ed0cfd91edc3a2b23a34a8ecd9db38baa12b52fa/packages/discord.js/src/structures/AnonymousGuild.js#L69-L76

Both an `InviteGuild` and a `Guild` can receive the animated hash.

**Review by Jiralite — CHANGES_REQUESTED** [review]

**Comment by Jiralite** [comment]
Upstream got merged. Are you able to continue, @iShibi?

**Comment by iShibi** [comment]
> Upstream got merged. Are you able to continue, @iShibi?

it's been a long gap, you can takeover if it's fine with you

**Comment by Jiralite** [comment]
I've looked over what needs to be done to support animated guild banners in discord.js.

For some reason, the `AnonymousGuild`'s `bannerURL()` method already supports dynamic hashes:

https://github.com/discordjs/discord.js/blob/402514ff323ccf1f8c95d295f044cf0bb5547c2e/packages/discord.js/src/structures/AnonymousGuild.js#L78-L85

https://github.com/discordjs/discord.js/blob/402514ff323ccf1f8c95d295f044cf0bb5547c2e/packages/discord.js/typings/index.d.ts#L268

All that's left is to bump discord-api-types to 0.31.0 so the typings are updated for the new guild feature. However, this is already being done in #7634, so it looks like no pull request is needed for this and support will naturally come!

**Comment by SpaceEEC** [comment]
See comment above ^

**Pull request closed without merging**